assets/test_chambers/test_chamber_09/test_chamber_09.yaml

```diff
@@ -23,5 +25,20 @@ cutscenes:
     - play_animation pit_1 pit_1_down -1
     - delay 3
     - goto pit_loop
+  CHECK_TRAPPED:
+    - wait_for_signal cube_on_piston 210
```

Comment:
This is getting a bit complicated with chains of `wait_for_signal` steps to coordinate between multiple cutscenes. And the piston check isn't disabled after the chamber is solved the intended way.

I'm not finished my cutscene documentation, but `stop_cutscene` (currently undocumented) - along with operators - would be more succinct and correct.

I.e.,
```yaml
cutscenes:
  ...

  CHECK_CUBE_ON_PISTON:
    - wait_for_signal cube_trapped 210
    - stop_cutscene CHECK_CUBE_OFF_BUTTON
    ...
    - set_signal escape_hatch
  
  CHECK_CUBE_OFF_BUTTON:
    - wait_for_signal open_door_cube
    - wait_for_signal not_cube_near_button
    - stop_cutscene CHECK_CUBE_ON_PISTON
    ...
    - set_signal escape_hatch

operators:
  - open_door = open_door or escape_hatch
  - unsolved = not exit_activate
  - cube_trapped = cube_on_piston and unsolved
  - not_cube_near_button = not cube_near_button
```

Can you change to something like this?

Reply:
Nice, that makes things a bit easier indeed. With this in mind, I think it'd be more conclusive to use `stop_cutscene CHECK_CUBE_ON_PISTON` in the `INTRO` cutscene instead of using `unsolved`.